packages/charts/react-charting/src/utilities/utilities.ts

```diff
@@ -158,35 +161,59 @@ export interface IYAxisParams {
  * @export
  * @param {IXAxisParams} xAxisParams
  */
-export function createNumericXAxis(xAxisParams: IXAxisParams, chartType: ChartTypes, culture?: string) {
+export function createNumericXAxis(
+  xAxisParams: IXAxisParams,
+  tickParams: ITickParams,
+  chartType: ChartTypes,
+  culture?: string,
+) {
   const {
     domainNRangeValues,
     showRoundOffXTickValues = false,
     xAxistickSize = 6,
     tickPadding = 10,
-    xAxisCount = 6,
+    xAxisCount,
     xAxisElement,
+    hideTickOverlap,
   } = xAxisParams;
   const xAxisScale = d3ScaleLinear()
     .domain([domainNRangeValues.dStartValue, domainNRangeValues.dEndValue])
     .range([domainNRangeValues.rStartValue, domainNRangeValues.rEndValue]);
   showRoundOffXTickValues && xAxisScale.nice();
 
+  let tickCount = xAxisCount ?? 6;
+  const tickFormat = (domainValue: NumberValue, _index: number) => {
+    if (tickParams.tickFormat) {
+      return d3Format(tickParams.tickFormat)(domainValue);
+    }
+    const xAxisValue = typeof domainValue === 'number' ? domainValue : domainValue.valueOf();
+    return convertToLocaleString(xAxisValue, culture) as string;
+  };
+  if (hideTickOverlap && typeof xAxisCount === 'undefined') {
+    const longestLabelWidth =
+      calculateLongestLabelWidth(xAxisScale.ticks().map(tickFormat), '[class^="xAxis-"] text') + 20;
+    const [start, end] = xAxisScale.range();
+    tickCount = Math.max(1, Math.floor(Math.abs(end - start) / longestLabelWidth));
```

Comment:
>1 [](http://example.com/codeflow?start=25&length=1)

if only 1 tick is shown, which one is it -  the middle one?

Reply:
No, it is generally close to the xMax.